src/main/java/com/ericsson/ei/handlers/ObjectHandler.java

```diff
@@ -187,39 +187,20 @@ public String findObjectById(String id) {
      * sets the time to live value.
      * @return document
      * */
-    public BasicDBObject prepareDocumentForInsertion(String id, String object) {
-        BasicDBObject document = new BasicDBObject();
+    public BasicDBObject prepareDocumentForInsertion(String id, String object) {        
+        BasicDBObject document = BasicDBObject.parse(object);
         document.put("_id", id);
-        document.put("aggregatedObject", BasicDBObject.parse(object));
-
-        if (getTtl() > 0) {
-            try {
-                document.put("Time", DateUtils.getDate());
-            } catch (ParseException e) {
-                LOGGER.error("Failed to attach date to document.", e);
+        try {
+            if (getTtl() > 0) {               
+                document.put("Time", DateUtils.getDate());                
```

Comment:
I was thinking, what happens if the user defines a key called "Time" in their rules, we will overwrite this value when we insert a document into the database. If there is no possibility to check and validate the rules, then we have to clearly document a warning to users, saying which keys are reserved.

Reply:
Should be added in docs